clang/lib/Serialization/ASTWriter.cpp

```diff
@@ -3018,13 +3019,19 @@ void ASTWriter::WriteSubmodules(Module *WritingModule) {
     SourceLocationEncoding::RawLocEncoding DefinitionLoc =
         getRawSourceLocationEncoding(getAdjustedLocation(Mod->DefinitionLoc));
 
+    ModuleMap &ModMap = PP->getHeaderSearchInfo().getModuleMap();
+    FileID InferredFID =
+        Mod->IsInferred ? ModMap.getModuleMapFileIDForUniquing(Mod) : FileID();
+    int Inferred = getAdjustedFileID(InferredFID).getOpaqueValue();
```

Comment:
Nit: I would name this something less generic like "InferredAdjustedFID"

Reply:
I don't want to make it that long, since I'd need to wrap the entire `Record` initializer list and cause downstream merge conflict. In the end, I renamed `InferredFID` to `UnadjustedInferredFID` and `Inferred` to `InferredFID`.